exercises/practice/bank-account/.meta/example.cpp

```diff
@@ -0,0 +1,54 @@
+#include "example.h"
+
+#include <stdexcept>
+
+void Bankaccount::open() {
+    if (open_) {
+        throw std::logic_error{"account already open"};
+    }
+    open_ = true;
+    balance_ = 0;
+}
+
+void Bankaccount::deposit(int amount) {
+    std::lock_guard<std::mutex> guard(mutex_);
+    check_account_open();
+    check_amount_greater_zero(amount);
+
+    balance_ += amount;
+}
+
+void Bankaccount::withdraw(int amount) {
+    std::lock_guard<std::mutex> guard(mutex_);
+
+    check_account_open();
+    check_amount_greater_zero(amount);
+
+    if (amount > balance_) {
+        throw std::logic_error{"amount must be less than current balance"};
+    }
```

Comment:
The condition and the error message are slightly contradicting: Is `amount == balance_` allowed?